### ISSUE TITLE
Version number not auto generated correctly on built package.

### ISSUE THREAD

**Issue by KFR42** [NuGet/Home#3659]
My assembly info has:

`[assembly: AssemblyInformationalVersion("1.0.*")]`

But the package is built as version 1.0.0.  If I remove that then the package builder picks up:

`[assembly: AssemblyVersion("1.0.*")]`

Which results in the package number being correctly auto generated.

This was working correctly in the last stable build.

**Comment by rrelyea** [comment]
@KFR42 - what version are you using now? What version last worked as your desire?

**Comment by rrelyea** [comment]
@KFR42 - Are you able to answer that question?

**Comment by KFR42** [comment]
Sorry it took me a while to get back to you.

I was previously using version 3.4.4 and am now testing with version 3.5.

As far as I can tell, the replacement tokens in the nuspec files do not work in many cases in the new version.  As well as the versioning issue I mentioned above.

My project uses an assembly.info file as well as several .info files which are common to all projects and added in as links (although, copying the values into the main assembly.info file makes no difference).

My assembly.info:

[assembly: AssemblyTitle("Core")]

[assembly: Guid("566590b2-acdc-400c-898d-7e86ab880eba")]

My shared assembly info file linked into the properties directory of my project:

[assembly: AssemblyCompany("My company Ltd.")]
[assembly: AssemblyProduct("My product")]
[assembly: AssemblyCopyright("Copyright © 2016")]
[assembly: AssemblyTrademark("")]
[assembly: AssemblyCulture("")]

[assembly: AssemblyConfiguration("Debug")]
[assembly: AssemblyDescription("Flavor=Debug")]

[assembly: ComVisible(false)]

[assembly: AssemblyVersion("1.0.*")]

[assembly: AssemblyInformationalVersion("1.0.*")] 

My nuspec:

`<?xml version="1.0"?>
<package >
  <metadata>
    <id>$id$</id>
    <version>$version$</version>
    <title>$title$</title>
    <authors>$author$</authors>
    <owners>$author$</owners>
    <projectUrl>http://PROJECT_URL_HERE_OR_DELETE_THIS_LINE</projectUrl>
    <iconUrl>http://ICON_URL_HERE_OR_DELETE_THIS_LINE</iconUrl>
    <requireLicenseAcceptance>false</requireLicenseAcceptance>
    <description>Package that defines core system logic.</description>
    <releaseNotes>Initial release of this package.</releaseNotes>
  </metadata>
</package>`

With version 3.4.4 of NuGet this packed correctly.  The right information was in the nuspec in the package (including author) and the version was automatically generated as 1.0.day value.time value.  With version 3.5, downloaded as of yesterday, this fails to pack saying authors is required.  The same result if I move AssemblyCompany into Assembly.Info.  If I hard code the author into the nuspec file instead it packs, but the version number is 1.0.0.

It seems to me that this whole area is a bit broken in the new release.  Unless there's some change to the way I should be doing this.  Here is my packing batch:

nuget pack .\src\Core\Core.csproj -Symbols -IncludeReferencedProjects -OutputDirectory .\nupkgs

**Comment by chrisnelson** [comment]
FYI, I was experiencing the same issue and I wrote up a whole post outlining my situation. Of course, I tried one more thing before posting and finally figured out my issue. I am posting in the event it might be helpful for the original poster or others.

I was using version 3.3.0 on our build server. I ran into an issue with a project referencing the latest AutoMapper package because that has targets/frameworks not defined in nuget 3.3. The solution was to upgrade to 3.4+. We have a single instance of the nuget.exe file on our build server, and I didn't want to break other builds so I downloaded the new nuget.exe file (3.5) and placed it side-by-side, renamed to nuget.3.5.exe. I then tried packing the project referencing AutoMapper and that error went away, but token replacement seemed to stop all together. I did notice, after adding the -verbosity Detailed switch, that there was a WARNING saying it couldn't read the metadata of the source project DLL. I didn't think much (it is only a WARNING!) but after hours of banging my head on the wall, I listened to the warning which said to get more details, enable Fusion logging in the registry. I did so, and it became clear what was happening: **you can not rename the nuget.exe file to anything else and expect it to work!** I didn't dig into why, but it was trying to load the nuget.exe file (not nuget.3.5.exe) which wasn't the same version so an ImageLoadException occurred. 

LONG story short: **don't try to rename the nuget.exe file, it needs to have that exact filename for the token replacement code to work.**

I'm not sure if the original poster was trying to use nuget.exe with a different filename, but worth looking into. 

Hope this might help someone else out!

**Comment by KFR42** [comment]
Alas, no. I am running it as nuget.exe. i have other renamed nuget exes in the same directorym could they be affecting it?

**Comment by chrisnelson** [comment]
Probably won't matter, but doesn't hurt to remove them or isolate the project file you are updating to have just the 3.5 exe file. Try adding the -verbosity Detailed to the call and see if there is anything interesting. With me, I had a warning, that when I drilled into, said it couldn't read the DLL's metadata so that is why the token replacement wasn't happening. Curious if you are having the same issue (but maybe for a different cause).

**Comment by KFR42** [comment]
This is what I get with detailed logging.  No warnings, just the validation error that it needs the authors.  I tried removing other renamed nuget.exe's from the folder and also tried removing some of the options from the pack and running it from within the project folder itself.  Same error every time.

C:\TFS\Core\vFuture\Core>nuget pack .\src\Core\Core.csproj -Symbols -IncludeReferencedProjects -OutputDirectory .\nupkgs -verbosity detailed
NuGet Version: 3.5.0.1938
Attempting to build package from 'Core.csproj'.
MSBuild auto-detection: using msbuild version '14.0' from 'C:\Program Files (x86)\MSBuild\14.0\bin'. Use option -MSBuildVersion to force nuget to use a specific version of MSBuild.
Packing files from 'C:\TFS\Core\vFuture\Core\bin\Debug'.
Using 'Core.nuspec' for metadata.
System.Exception: Authors is required.
   at NuGet.Packaging.Manifest.Validate(Manifest manifest)
   at NuGet.Packaging.Manifest.ReadFrom(Stream stream, Func`2 propertyProvider, Boolean validateSchema)
   at NuGet.CommandLine.ProjectFactory.ProcessNuspec(PackageBuilder builder, String basePath)
   at NuGet.CommandLine.ProjectFactory.CreateBuilder(String basePath, NuGetVersion version, String suffix, Boolean buildIfNeeded)
   at NuGet.Commands.PackCommandRunner.BuildFromProjectFile(String path)
   at NuGet.Commands.PackCommandRunner.BuildPackage(String path)
   at NuGet.Commands.PackCommandRunner.BuildPackage()
   at NuGet.CommandLine.PackCommand.ExecuteCommand()
   at NuGet.CommandLine.Command.ExecuteCommandAsync()
   at NuGet.CommandLine.Command.Execute()
   at NuGet.CommandLine.Program.MainCore(String workingDirectory, String[] args)

UPDATE: After some playing about, I think I have isolated the error.  If I remove the AssemblyInformationalVersion from AssemblyInfo, it starts working again.  It also works if I remove the '.*' from the value of that attribute.  Maybe that isn't a valid value for that attribute, but the error returned by nuget pack is certainly wrong. @rrelyea, please could you investigate.

**Comment by jainaashish** [comment]
@KFR42 @chrisnelson hey guys, is it still an issue with latest NuGet?

**Comment by rrelyea** [comment]
Closing, but please let us know if it is still an issue.
